include/acodec.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API header file for Audio Codec
+ *
+ * This file contains the Audio Codec APIs
+ */
+
+#ifndef __ACODEC_H__
+#define __ACODEC_H__
+
+#include <i2s.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * PCM audio sample rates
+ */
+typedef enum {
+	ACODEC_PCM_RATE_8K	= 8000,
+	ACODEC_PCM_RATE_16K	= 16000,
+	ACODEC_PCM_RATE_24K	= 24000,
+	ACODEC_PCM_RATE_32K	= 32000,
+	ACODEC_PCM_RATE_44P1K	= 44100,
+	ACODEC_PCM_RATE_48K	= 48000,
+	ACODEC_PCM_RATE_96K	= 96000,
+	ACODEC_PCM_RATE_192K	= 192000,
+} acodec_pcm_rate_t;
+
+/**
+ * PCM audio sample bit widths
+ */
+typedef enum {
+	ACODEC_PCM_WIDTH_16_BITS	= 16,
+	ACODEC_PCM_WIDTH_20_BITS	= 20,
+	ACODEC_PCM_WIDTH_24_BITS	= 24,
+	ACODEC_PCM_WIDTH_32_BITS	= 32,
+} acodec_pcm_width_t;
+
+/**
+ * Digital Audio Interface (DAI) type
+ */
+typedef enum {
+	ACODEC_DAI_TYPE_I2S,		/* I2S Interface */
+	ACODEC_DAI_TYPE_INVALID,	/* Other interfaces can be added here */
+} acodec_dai_type_t;
+
+/**
+ * Digital Audio Interface Configuration
+ * Configuration is dependent on DAI type
+ */
+typedef union {
+	struct i2s_config i2s;	/* I2S configuration */
+				/* Other DAI types go here */
+} acodec_dai_cfg_t;
+
+/**
+ * Codec configuration parameters
+ */
+struct acodec_cfg {
+	u32_t			mclk_freq;	/* MCLK input frequency in Hz */
+	acodec_dai_type_t	dai_type;	/* Digital interface type */
+	acodec_dai_cfg_t	dai_cfg;	/* DAI configuration info */
+};
+
+/**
+ * Codec output volume control range
+ * Volume range is from 0dB to -78dB in steps of 0.5dB
+ * Example"
+ * vol = -41 is -20.5dB
+ * vol = -156 is -78dB
+ */
+#define ACODEC_OUTPUT_VOLUME_MAX		0
```

Comment:
This depends on the hardware. In my opinion we should not specify such limits in generic header.

Reply:
@pizi-nordic Limits removed.

Reply:
Thank you.
What about: https://github.com/zephyrproject-rtos/zephyr/pull/7531#discussion_r188222333 (property-based interface)?

Reply:
Looks like I missed that one. Will address shortly.

Reply:
Please look through all comments, as there could be more missed items.

Reply:
@pizi-nordic Done.
@pizi-nordic @nashif By the way, I am unable to get used to this `acodec` prefix. What do you think of moving the audio related headers as `include/audio/codec.h` and `include/audio/dmic.h`
Refer to #8572